addons/delivery/delivery.py

```diff
@@ -199,12 +192,12 @@ def get_price(self, cr, uid, id, order, dt, context=None):
         return self.get_price_from_picking(cr, uid, id, total,weight, volume, quantity, context=context)
 
     def get_price_from_picking(self, cr, uid, id, total, weight, volume, quantity, context=None):
```

Comment:
get_price_from_picking method should be process on batch. so change method prototype . use ids instead of id.